File: src/domain/entities/session/SessionActivationState.ts
/**
 * Durable activation state of a mirrored session.
 * It represents onboarding progress independently from the live QR/code events.
 */
export enum SessionActivationState {
  Idle = 'idle',
  AwaitingQrCode = 'awaiting_qr_code',
  AwaitingPairingCode = 'awaiting_pairing_code',
  Completed = 'completed',
  Expired = 'expired',
  Failed = 'failed',
  Cancelled = 'cancelled',
}

export function parseSessionActivationState(value: string): SessionActivationState {
  switch (value) {
    case SessionActivationState.Idle:
      return SessionActivationState.Idle;
    case SessionActivationState.AwaitingQrCode:
      return SessionActivationState.AwaitingQrCode;
    case SessionActivationState.AwaitingPairingCode:
      return SessionActivationState.AwaitingPairingCode;
    case SessionActivationState.Completed:
      return SessionActivationState.Completed;
    case SessionActivationState.Expired:
      return SessionActivationState.Expired;
    case SessionActivationState.Failed:
      return SessionActivationState.Failed;
    case SessionActivationState.Cancelled:
      return SessionActivationState.Cancelled;
    default:
      throw new Error(`Unsupported session activation state "${value}".`);
  }
}

File: sdks/java/src/main/java/com/digows/whatsappgateway/session/SessionActivationState.java
package com.digows.whatsappgateway.session;

import com.fasterxml.jackson.annotation.JsonCreator;
import com.fasterxml.jackson.annotation.JsonValue;

public enum SessionActivationState
{
  IDLE("idle"),
  AWAITING_QR_CODE("awaiting_qr_code"),
  AWAITING_PAIRING_CODE("awaiting_pairing_code"),
  COMPLETED("completed"),
  EXPIRED("expired"),
  FAILED("failed"),
  CANCELLED("cancelled");

  private final String wireValue;

  SessionActivationState(String wireValue)
  {
    this.wireValue = wireValue;
  }

  @JsonValue
  public String getWireValue()
  {
    return wireValue;
  }

  @JsonCreator
  public static SessionActivationState fromWireValue(String wireValue)
  {
    for (SessionActivationState value : values())
    {
      if (value.wireValue.equals(wireValue))
      {
        return value;
      }
    }

    throw new IllegalArgumentException("Unsupported session activation state \"" + wireValue + "\".");
  }
}
